staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/etcd.go

```diff
@@ -30,8 +31,11 @@ import (
 	"k8s.io/apiserver/pkg/storage"
 	storageerr "k8s.io/apiserver/pkg/storage/errors"
 	"k8s.io/apiserver/pkg/util/dryrun"
+	printersinternal "k8s.io/kubernetes/pkg/printers/internalversion"
+	printerstorage "k8s.io/kubernetes/pkg/printers/storage"
```

Comment:
things inside the staging directory are not permitted to depend on things under `k8s.io/kubernetes/...`

@seans3 @smarterclayton, should the base table generator be moved to k8s.io/apiserver (and the handlers for specific types be placed with the apiserver responsible for those types)?

Reply:
Since I can't import from `k8s.io/kubernetes/..` from the staging directory. How will I get access to the `TableGenerator` and and `TableConverter` defined in the `printers` and `printerstorage` packages? Assuming I implement my `customresourcedefinitionHandlerFunction` in `k8s.io/apiserver`  instead of in the already existing `addHandlersFunction` in `printers.go`

Reply:
I'm proposing moving the TableGenerator/TableConvertor to `k8s.io/apiserver` (where it logically needs to live for arbitrary API servers to implement this), and implementing this particular handler within `k8s.io/apiextensions-apiserver`

Reply:
May I proceed to raise a PR for this?

Reply:
Yes, it's about time we move the TableGenerator to `k8s.io/apiserver`, @MartinKaburu feel free to open a PR, but a separate from this one. Once we move the TableGenerator we can proceed with this.